wc-src/vf-voice.js

```diff
@@ -0,0 +1,83 @@
+import Vex from '../src/index.js';
+
+const template = document.createElement('template');
+template.innerHTML = `
+`;
+
+function concat(a, b) {
+  return a.concat(b);
+}
+
+export class VFVoice extends HTMLElement {
+  constructor() {
+    super();
+
+    this.attachShadow({ mode:'open' });
+    this.shadowRoot.appendChild(document.importNode(template.content, true));
+
+    // Defaults
+    this.stem = 'up';
+    this.autoBeam = false;
+    this.notes = [];
+    this.beams = [];
+
+    this._vf = undefined;
+    this._score = undefined;
+  }
+
+  connectedCallback() {
+    this.stem = this.getAttribute('stem') || this.stem;
+    this.autoBeam = this.hasAttribute('autoBeam');
+    this.notesText = this.textContent.trim();
+
+    const getFactoryEvent = new CustomEvent('getFactory', { bubbles: true });
+    this.dispatchEvent(getFactoryEvent);
+
+    const getScoreEvent = new CustomEvent('getScore', { bubbles: true });
+    this.dispatchEvent(getScoreEvent);
+  }
+
+  set vf(value) {
+    this._vf = value;
+    this.createNotes();
+  }
+
+  set score(value) {
+    this._score = value;
+    this.createNotes();
+  }
+
+  createNotes() {
+    if (this._vf && this._score) {
```

Comment:
It just occurred to me: Instead of having vf-score set two properties, what if it called a method instead?

vf-voice etc could define `setup(factory, score)`. That way we wouldn't have to call createNotes twice and have this "both properties are defined" guard. `setup(factory, score)` could call straight to `createNotes` with no worries.

Reply:
With what you said above, ignore this 👍